### PR TITLE
[docs] mention extensions in relevant tutorials

### PR DESCRIPTION
**Pull request by odow**
x-ref https://github.com/jump-dev/JuMP.jl/pull/3623#issuecomment-1852127896

I think the rule for this is that the extension must be part of the `Extensions` tab of the JuMP docs, and we must link there. That means if we ever remove an extension from the docs, it'll also catch removing these links.

### CODE REVIEW

File: docs/src/tutorials/nonlinear/rocket_control.jl [hunk 1]
```diff
@@ -12,6 +12,10 @@
 
 # The example is an optimal control problem of a nonlinear rocket.
 
+# !!! info
+#     The JuMP extension [InfiniteOpt.jl](/packages/InfiniteOpt) can also be
```
Comment (odow) [inline docs/src/tutorials/nonlinear/rocket_control.jl:16]:
```suggestion
#     The JuMP extension [InfiniteOpt.jl](../../../packages/InfiniteOpt) can also be
```

File: docs/src/tutorials/nonlinear/space_shuttle_reentry_trajectory.jl [hunk 2]
```diff
@@ -31,6 +31,10 @@
 # ["Practical Methods for Optimal Control and Estimation Using Nonlinear Programming"](https://epubs.siam.org/doi/book/10.1137/1.9780898718577),
 # by John T. Betts.
 
+# !!! info
+#     The JuMP extension [InfiniteOpt.jl](/packages/InfiniteOpt) can also be
```
Comment (odow) [inline docs/src/tutorials/nonlinear/space_shuttle_reentry_trajectory.jl:35]:
```suggestion
#     The JuMP extension [InfiniteOpt.jl](../../../packages/InfiniteOpt) can also be
```

File: docs/src/tutorials/nonlinear/rocket_control.jl [hunk 3]
```diff
@@ -12,6 +12,10 @@
 
 # The example is an optimal control problem of a nonlinear rocket.
 
+# !!! info
+#     The JuMP extension [InfiniteOpt.jl](../../../packages/InfiniteOpt) can also be
```
Comment (odow) [inline docs/src/tutorials/nonlinear/rocket_control.jl:16]:
```suggestion
#     The JuMP extension [InfiniteOpt.jl](../../packages/InfiniteOpt) can also be
```

File: docs/src/tutorials/nonlinear/space_shuttle_reentry_trajectory.jl [hunk 4]
```diff
@@ -31,6 +31,10 @@
 # ["Practical Methods for Optimal Control and Estimation Using Nonlinear Programming"](https://epubs.siam.org/doi/book/10.1137/1.9780898718577),
 # by John T. Betts.
 
+# !!! info
+#     The JuMP extension [InfiniteOpt.jl](../../../packages/InfiniteOpt) can also be
```
Comment (odow) [inline docs/src/tutorials/nonlinear/space_shuttle_reentry_trajectory.jl:35]:
```suggestion
#     The JuMP extension [InfiniteOpt.jl](../../packages/InfiniteOpt) can also be
```

**Comment by odow** [comment]
I never get these relative links correct

File: docs/src/tutorials/nonlinear/rocket_control.jl [hunk 5]
```diff
@@ -12,6 +12,10 @@
 
 # The example is an optimal control problem of a nonlinear rocket.
 
+# !!! info
+#     The JuMP extension [InfiniteOpt.jl](../../packages/InfiniteOpt) can also be
```
Comment (odow) [inline docs/src/tutorials/nonlinear/rocket_control.jl:16]:
```suggestion
#     The JuMP extension [InfiniteOpt.jl](./../../packages/InfiniteOpt) can also be
```

File: docs/src/tutorials/nonlinear/space_shuttle_reentry_trajectory.jl [hunk 6]
```diff
@@ -31,6 +31,10 @@
 # ["Practical Methods for Optimal Control and Estimation Using Nonlinear Programming"](https://epubs.siam.org/doi/book/10.1137/1.9780898718577),
 # by John T. Betts.
 
+# !!! info
+#     The JuMP extension [InfiniteOpt.jl](../../packages/InfiniteOpt) can also be
```
Comment (odow) [inline docs/src/tutorials/nonlinear/space_shuttle_reentry_trajectory.jl:35]:
```suggestion
#     The JuMP extension [InfiniteOpt.jl](/../../packages/InfiniteOpt) can also be
```

File: docs/src/tutorials/nonlinear/nested_problems.jl [hunk 7]
```diff
@@ -32,6 +32,10 @@
 # For a simpler example of writing a user-defined operator,
 # see the [User-defined Hessians](@ref) tutorial.
 
+# !!! info
+#     The JuMP extension [BilevelJuMP.jl](@ref) can also be used to model and
```
Comment (odow) [inline docs/src/tutorials/nonlinear/nested_problems.jl:36]:
```suggestion
#     The JuMP extension [BilevelJuMP.jl](./../packages/BilevelJuMP) can also be used to model and
```

File: docs/src/tutorials/nonlinear/rocket_control.jl [hunk 8]
```diff
@@ -12,6 +12,10 @@
 
 # The example is an optimal control problem of a nonlinear rocket.
 
+# !!! info
+#     The JuMP extension [InfiniteOpt.jl](./../../packages/InfiniteOpt) can also be
```
[the remark this answers is not in the document]
Comment (mortenpi) [reply docs/src/tutorials/nonlinear/rocket_control.jl:16]: The links should point to the actual `.md` file I think:

```suggestion
#     The JuMP extension [InfiniteOpt.jl](../../packages/InfiniteOpt.md) can also be
```

File: docs/src/tutorials/nonlinear/nested_problems.jl [hunk 9]
```diff
@@ -32,6 +32,10 @@
 # For a simpler example of writing a user-defined operator,
 # see the [User-defined Hessians](@ref) tutorial.
 
+# !!! info
+#     The JuMP extension [BilevelJuMP.jl](./../packages/BilevelJuMP) can also be used to model and
```
Comment (odow) [inline docs/src/tutorials/nonlinear/nested_problems.jl:36]:
```suggestion
#     The JuMP extension [BilevelJuMP.jl](../../packages/BilevelJuMP.md) can also be used to model and
```

File: docs/src/tutorials/nonlinear/space_shuttle_reentry_trajectory.jl [hunk 10]
```diff
@@ -31,6 +31,10 @@
 # ["Practical Methods for Optimal Control and Estimation Using Nonlinear Programming"](https://epubs.siam.org/doi/book/10.1137/1.9780898718577),
 # by John T. Betts.
 
+# !!! info
+#     The JuMP extension [InfiniteOpt.jl](/../../packages/InfiniteOpt) can also be
```
Comment (odow) [inline docs/src/tutorials/nonlinear/space_shuttle_reentry_trajectory.jl:35]:
```suggestion
#     The JuMP extension [InfiniteOpt.jl](../../packages/InfiniteOpt.md) can also be
```

File: docs/src/tutorials/nonlinear/rocket_control.jl (hunk 8 above)
[the remark this answers is not in the document]
Comment (odow) [reply docs/src/tutorials/nonlinear/rocket_control.jl]: D'oh. I was thinking of HTML links.

[the remark this answers is not in the document]
Comment (odow) [reply docs/src/tutorials/nonlinear/rocket_control.jl]: Which I guess explains why I always have trouble with them

**Pull request merged**